src/Compilers/CSharp/Portable/Binder/Binder_Await.cs

```diff
@@ -431,5 +432,102 @@ private static bool HasOptionalOrVariableParameters(MethodSymbol method)
 
             return false;
         }
+
+        /// <summary>
+        /// Returns the appropriate AsyncMethodBuilder based on whether the
+        /// return type is void or a non-generic or generic task-like type. The
+        /// caller is responsible for ensuring this is used for async methods only.
+        /// </summary>
+        internal AsyncMethodBuilder GetAsyncMethodBuilder(CSharpSyntaxNode syntax, TypeSymbol returnType)
```

Comment:
Could you explain conceptually why GetAsyncMethodBuilder is inside "binder", specifically the part of binder that's responsible for turning Await into a BoundAwaitExpression?

Because the binding for await expressions is completely unrelated to async-method-builder, I think...?